### DISCUSSION TITLE
Heavy pages files

### DISCUSSION THREAD

**Discussion by testdgb** [vercel/next.js#48755] [Help]
### Summary

Hey there, I'm currently developing a website on Next.js and I've noticed that there are 3 files that are quite heavy by default. Even when I create a new empty page, the entire page size is almost 2MB without adding anything. The files that contribute to this are main.js (1.1MB), app.js (0.46MB), and inpage.js (0.32MB).

How can I fix this issue? Also, how can I determine what is causing these files to be so large?

<img width="483" alt="Screenshot 2023-04-23 at 19 18 49" src="https://user-images.githubusercontent.com/122317690/233851744-cbd72f3e-610f-49c3-b52c-20fa23ff9552.png">

**Comment by icyJoseph** [comment]
Hi,

Could you run https://pagespeed.web.dev/ on your site and post a link back to the `Tree View`? If you have a link to a repository that'd help too.

I also see that your screenshot is from a development run, do you see the same in production mode?

To see what the issue could be, use the next bundle analyzer: https://www.npmjs.com/package/@next/bundle-analyzer

**Reply by testdgb** [reply]
Hey, thanks for your response!
My website is still under development, the screenshot is from my localhost:3000, and it's not yet online.
Also, what's the difference between development run to production mode in the Chrome browser?

**Reply by icyJoseph** [reply]
In development mode, there's a lot of supporting code to give you things such as Hot Module Reloading, warnings and extremely detailed error messages. Code is also not fully minified.

- https://overreacted.io/how-does-the-development-mode-work/

It's not only Next.js or other React frameworks, but a very extensive practice in this industry.

Using the bundle analyzer you can have a mire accurate view of your actual payloads.

**Reply by testdgb** [reply]
Oh okay! Thank you for letting me know!

**Comment by timneutkens** [comment]
You should run `next build` and then `next start` to get the production bundle size. The development bundle is larger because it does not get minified / tree shaken, it also includes additional code for e.g. Fast Refresh and such.

**Reply by testdgb** [reply]
Good to know. Thank you! :)
